cdist/core/cdist_type.py

```diff
@@ -204,11 +204,13 @@ def parameter_defaults(self):
             try:
                 defaults_dir = os.path.join(self.absolute_path, "parameter", "default")
                 for name in os.listdir(defaults_dir):
-                    with open(os.path.join(defaults_dir, name)) as fd:
-                        defaults[name] = fd.read().strip()
+                    try:
+                        with open(os.path.join(defaults_dir, name)) as fd:
+                            defaults[name] = fd.read().strip()
+                    except EnvironmentError:
+                        pass  # Swallow errors raised by open() or read()
             except EnvironmentError:
```

Comment:
Hello @acatton! First of all thanks for your PR. 

I am returining to work from a short break and try to wrap my head around what you did. Afaics, the outer try...except block catches the same exception as you have done in the inner, so your change should not make a difference.

Can you explain why this prevents the problem from happening?

I wonder if it makes sense to actually stat() before and only read the file, if it is a normal file and then to read it.

Btw, I have just seen that EnvironmentError vanished in python 3.3:

Changed in version 3.3: EnvironmentError, IOError, WindowsError, VMSError, socket.error, select.error and mmap.error have been merged into OSError.
from https://docs.python.org/3/library/exceptions.html